CMakeLists.txt

```diff
@@ -337,6 +337,26 @@ endif (WIN32)
 
 endif (OVERRIDE_CMAKE_CXX_FLAGS)
 
+OPTION(CMAKE_ENABLE_CODE_COVERAGE "Enable code coverage" OFF)
+
+if(CMAKE_ENABLE_CODE_COVERAGE)
+
+  string(TOUPPER ${CMAKE_BUILD_TYPE} UPPERCASE_CMAKE_BUILD_TYPE)
+  if(NOT UPPERCASE_CMAKE_BUILD_TYPE STREQUAL DEBUG)
```

Comment:
This won't work as you expect for generating VS projects on Windows or any other multi-target generator. You cannot test for the build type in the CMakeLists.txt files, since the CMake passes through the CMakeLists.txt just once for all the build types.